internal/services/containers/kubernetes_cluster_resource.go

```diff
@@ -1286,10 +1286,10 @@ func resourceKubernetesCluster() *pluginsdk.Resource {
 			},
 
 			"public_network_access_enabled": {
-				Type:     pluginsdk.TypeBool,
-				Optional: true,
-				Default:  true,
-				ForceNew: true,
+				Type:       pluginsdk.TypeBool,
+				Optional:   true,
+				Default:    true,
+				Deprecated: "`public_network_access_enabled` is not functional, will not be passed to the API",
```

Comment:
can we do a better job of explaining why this is being removed? and link to any public information?

Reply:
Hi @katbyte , we got confirmation from the service team that this function is not complete yet and they're working on it.

And there's no public doc about this field.

Reply:
Here is an [explanation ](https://github.com/Azure/AKS/issues/3690#issuecomment-1590425190)provided by AKS team suggesting this property is not fully functional ready.

Reply:
Recently AKS adds a new validation to prevent users creating clusters with "public_network_access_enabled=false" and UDR(user defined routing). 

Without this PR, when users create new cluster with UDR, they must change "public_network_access_enabled" to true or remove "public_network_access_enabled=false" from this config. This PR make it easier for users to adopt this validation, users don't need to change their config, and the existing clusters won't be affected.